tests/layer_tests/pytorch_tests/test_compressed_mm.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (C) 2018-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+import pytest
+import torch
+
+from pytorch_layer_test_class import PytorchLayerTest
+
+
+def pack_uint4(tensor):
+    packed_tensor = tensor.contiguous()
+    packed_tensor = packed_tensor.reshape(-1, 2)
+    packed_tensor = torch.bitwise_and(packed_tensor[..., ::2], 15) | packed_tensor[..., 1::2] << 4
+    return packed_tensor
+
+
+def unpack_uint4(packed_tensor):
+    return torch.stack((torch.bitwise_and(packed_tensor, 15), torch.bitwise_right_shift(packed_tensor, 4)), dim=-1)
+
+
+def pack_int4(tensor):
+    tensor = tensor + 8
+    return pack_uint4(tensor.type(torch.uint8))
+
+
+def unpack_int4(packed_tensor):
+    t = unpack_uint4(packed_tensor)
+    return t.type(torch.int8) - 8
+
+
+def decompress_asymmetric(input, scale, zero_point):
+    input = input.type(dtype=scale.dtype)
+    zero_point = zero_point.type(dtype=scale.dtype)
+    decompressed_input = (input - zero_point) * scale
+    return decompressed_input
+
+
+def decompress_symmetric(input, scale):
+    input = input.type(dtype=scale.dtype)
+    decompressed_input = input * scale
+    return decompressed_input
+
+
+class TestMatMulU4Weights(PytorchLayerTest):
+    rng = np.random.default_rng(seed=123)
+
+    def _prepare_input(self):
+        return np.round(5.00 * self.rng.random([2, 4], dtype=np.float32) - 2.50, 4)
+
+    def create_model(self, compressed_weight, scale, zero_point, weight_shape):
+        class aten_mm_u4(torch.nn.Module):
+            def __init__(self, compressed_weight, scale, zero_point, weight_shape):
+                super(aten_mm_u4, self).__init__()
+                self.compressed_weight_shape = compressed_weight.shape
+                self.packed_weight = torch.nn.Parameter(pack_uint4(compressed_weight), requires_grad=False)
+
+                self.register_buffer("_scale", scale.type(dtype=torch.float16))
+
+                self.zero_point_shape = zero_point.shape
+                self.register_buffer("_zero_point", pack_uint4(zero_point))
+
+                self.weigh = weight_shape
+
+            def forward(self, x):
+                compressed_weight = unpack_uint4(self.packed_weight)
```

Comment:
Could you make a reference to nncf code where this pattern is created? Is it possible to use nncf pattern here instead of creating it from operations?

Reply:
Link to the NNCF: https://github.com/openvinotoolkit/nncf/blob/c1d9fd753cc405ce52ebe9097b351b29736a582c/nncf/torch/quantization/layers.py#L1148

I don't think adding nncf dependency to openvino pre-commit tests is a good idea. But I would like to add nncf models to hub tests in the next PR.

Reply:
Okay, no need to add dependency, but please add a comment that this patter reflect nncf one, and add the link to code

Reply:
Done, thanks!